src/Bitmap/BitmapFile.cpp

```diff
@@ -35,16 +35,36 @@ void BitmapFile::VerifyIndexedPaletteSizeDoesNotExceedBitCount(uint16_t bitCount
 
 void BitmapFile::VerifyPixelSizeMatchesImageDimensionsWithPitch() const
 {
-	BitmapFile::VerifyPixelSizeMatchesImageDimensionsWithPitch(imageHeader.bitCount, imageHeader.width, imageHeader.height, pixels.size());
+	VerifyPixelSizeMatchesImageDimensionsWithPitch(imageHeader.bitCount, FindPitch(), imageHeader.height, pixels.size());
 }
 
-void BitmapFile::VerifyPixelSizeMatchesImageDimensionsWithPitch(uint16_t bitCount, int32_t width, int32_t height, std::size_t pixelsWithPitchSize)
+void BitmapFile::VerifyPixelSizeMatchesImageDimensionsWithPitch(uint16_t bitCount, std::size_t pitch, int32_t height, std::size_t pixelsWithPitchSize)
 {
-	if (pixelsWithPitchSize != ImageHeader::CalculatePitch(bitCount, width) * std::abs(height)) {
+	if (pixelsWithPitchSize != pitch * std::abs(height)) {
 		throw std::runtime_error("The size of pixels does not match the image's height time pitch");
 	}
 }
 
+std::size_t BitmapFile::FindPitch() const
+{
+	return FindPitch(imageHeader.width, imageHeader.height, pixels.size());
+}
+
+std::size_t BitmapFile::FindPitch(std::size_t width, std::size_t height, std::size_t pixelCount)
+{
+	if (pixelCount % height != 0) {
+		throw std::runtime_error("Unable to calculate a valid pitch based on height and pixel count");
+	}
+
+	const std::size_t pitch = pixelCount / height;
```

Comment:
This hack of mine is causing the unit tests to fail. They happen to be fairly robust for this small section of the code compared to others.

The next step is probably to figure out a proper bit twiddle or calculation that determines pitch without assuming it is a 4 byte interval. @DanRStevens, feel free to chime in if you have something up your sleeve.

-Brett

Reply:
Pitch is either assumed, or given. It's not generally calculated (other than in an assumption, such as rounding the scanline byte width up to a multiple of 4 bytes). The problem with trying to calculate it, is the data source could always have chosen a larger pitch than is necessary.

I find it a little non-intuitive to use `pixelCount` here. When I think pixel could, I generally assume it's simply `width * height`, which is not particularly useful for calculating `pitch` if you already have the width. Plus, `pitch` is a byte measurement, not a pixel measurement. The `bitDepth` needs to be accounted for.

Reply:
If you are wanting to work on this branch feel free. I'm not going to pick it up again for a long time.

-Brett

Reply:
I'm not sure I quite have the context to pick this one up right now.